components/server/src/workspace/workspace-service.ts

```diff
@@ -119,8 +122,55 @@ export class WorkspaceService {
         return workspace;
     }
 
-    async getWorkspace(userId: string, workspaceId: string): Promise<Workspace> {
-        return this.doGetWorkspace(userId, workspaceId);
+    async getWorkspace(
+        userId: string,
+        workspaceId: string,
+        check: (workspace: Workspace, instance?: WorkspaceInstance) => Promise<void> = async () => {},
+    ): Promise<WorkspaceInfo> {
+        const workspace = await this.doGetWorkspace(userId, workspaceId);
+
+        const latestInstancePromise = this.db.findCurrentInstance(workspaceId);
+        await check(workspace);
+        const latestInstance = await latestInstancePromise;
+        await check(workspace, latestInstance);
+
+        return {
+            workspace,
+            latestInstance,
+        };
+    }
+
+    async getWorkspaces(
+        userId: string,
+        options: GitpodServer.GetWorkspacesOptions,
+        check: (workspace: Workspace, instance?: WorkspaceInstance) => Promise<void> = async () => {},
```

Comment:
same, i.e. don'T think we need to pass in the check function

Reply:
I just happened to notice that the old behavior was not a filter, but a "block any".
But you are right, we don't need to fix this now.